dali/pipeline/data/tensor_vector.h

```diff
@@ -375,6 +375,24 @@ class TensorVector {
     }
   }
 
+  void Swap(TensorVector<Backend> *tv) {
+    std::swap(state_, tv->state_);
+    std::swap(pinned_, tv->pinned_);
+    std::swap(tl_, tv->tl_);
+    std::swap(type_, tv->type_);
+    if (views_count_) {
+      tensors_.clear();
+      views_count_ = 0;
+    }
+    if (tv->views_count_) {
+      tv->tensors_.clear();
+      tv->views_count_ = 0;
+    }
```

Comment:
Why not just swap the views as well?

Reply:
Becasue views have a captured reference to the views_count_ . So when view is changed then it messes up with the views_count_  in the old tensor_vector.

Reply:
Reworked